wren-ai-service/Makefile

```diff
@@ -1,61 +1,243 @@
+		exit 1; \
+	fi
+
+check-poetry:
+	@echo "$(YELLOW)Checking Poetry installation...$(RESET)"
+	@if command -v poetry > /dev/null; then \
+		echo "$(BLUE)$(shell poetry --version) is already installed.$(RESET)"; \
+	else \
+		echo "$(RED)Poetry is not installed. You can install poetry by running the following command," \
+					" then adding Poetry to your PATH:"; \
+		echo "$(RED) curl -sSL https://install.python-poetry.org | python3 - --version 1.7.1$(RESET)"; \
+		echo "$(RED)More detail here: https://python-poetry.org/docs/#installing-with-the-official-installer$(RESET)"; \
+		exit 1; \
+	fi
+
+install-python-dependencies:
+	@echo "$(YELLOW)Installing dependencies...$(RESET)"
+	@poetry install
+	@echo "$(GREEN)Dependencies installed!$(RESET)"
+
+install-precommit-hooks:
+	@echo "$(YELLOW)Installing pre-commit hooks...$(RESET)"
+	@poetry run pre-commit install
+	@poetry run pre-commit run --all-files
+	@echo "$(GREEN)Pre-commit hooks installed!$(RESET)"
 
-run-qdrant:
-	docker run \
-	-p 6333:6333 \
-	-p 6334:6334 \
-	-d \
-	--name qdrant \
-	qdrant/qdrant:v1.7.4
+setup-envfile:
+	@echo "$(YELLOW)Setting up environment...$(RESET)"
+	@if [ -f ".env" ]; then \
+		echo "$(BLUE)environment file already exists.$(RESET)"; \
+	else \
+		echo "$(YELLOW)Creating .env file...$(RESET)"; \
+		cp .env.example .env; \
+		cp .env.dev.example .env.dev; \
+	fi
+
+start:
+	@echo "$(YELLOW)Starting all services...$(RESET)"
+	@poetry run python -m src.prepare_mdl_json
+	@$(MAKE) start-qdrant
+	@$(MAKE) start-wren-engine
+	@$(MAKE) start-ai
+	@echo "$(GREEN)All services started!$(RESET)"
+
+stop:
+	@echo "$(YELLOW)Stopping all services...$(RESET)"
+	@$(MAKE) stop-qdrant
+	@$(MAKE) stop-wren-engine
+	@$(MAKE) stop-ai
+	@echo "$(GREEN)All services stopped!$(RESET)"
+
+start-qdrant:
+	@echo "$(YELLOW)Starting Qdrant...$(RESET)"
+	@if docker ps --format '{{.Names}}' | grep 'qdrant' 1> /dev/null; then \
+		echo "$(BLUE)Qdrant is already running.$(RESET)"; \
+	else \
+		docker start qdrant > /dev/null 2>&1 || \
+		docker run -p 6333:6333 -p 6334:6334 -d --name qdrant qdrant/qdrant:v1.7.4; \
+	fi
+	@echo "$(GREEN)Qdrant started!$(RESET)"
 
 stop-qdrant:
-	docker stop qdrant && docker rm qdrant
+	@echo "$(YELLOW)Stopping Qdrant...$(RESET)"
+	@if docker ps --format '{{.Names}}' | grep 'qdrant' 1> /dev/null; then \
+		docker stop qdrant > /dev/null; \
+		echo "$(GREEN)Qdrant stopped!$(RESET)"; \
+	else \
+		echo "$(BLUE)Qdrant is not running.$(RESET)"; \
+	fi
+	@echo "$(YELLOW)Removing Qdrant...$(RESET)"
+	@if docker container ls -a --format '{{.Names}}' | grep 'qdrant' 1> /dev/null; then \
+		docker rm qdrant > /dev/null; \
+		echo "$(GREEN)Qdrant container removed!$(RESET)"; \
+	else \
+		echo "$(BLUE)Qdrant container does not exist.$(RESET)"; \
+	fi
 
-run-wren-engine:
-	docker compose -f ./src/eval/wren-engine/docker-compose.yml --env-file ./src/eval/wren-engine/.env up -d
+start-wren-engine:
```

Comment:
for the start-wren-engine part, users also need to prepare an etc folder first, and put the config.properties file inside and fill in environment variables.(generated from config.properties.example in the src/eval/wren-engine)

Reply:
I think it is a one-time behavior. I'd prefer to mention this in README instead of generating the config every time. or add another make file command to generate the config file. what do you think?

Reply:
sure. it's fine to me